atc/worker/pool.go

```diff
@@ -231,15 +232,27 @@ func (pool Pool) findWorkerForContainer(logger lager.Logger, owner db.ContainerO
 		return nil, nil, false, err
 	}
 
+	// When global-resources is enabled, a check-container may run on any team
```

Comment:
I think the check container will only run on either its team worker or a general worker but it will never run on another team's worker.

Reply:
I might not state clearly. What I really meant was that: say resource-1 of team-1, resource-2 of team-2 and resource-3 of team-3 has the same scope, and the 3 teams all have team workers. Then the 3 checks will run on all 3 team workers with the existing logic, which is actually a waste, as they share the same version history and acquiring the same lock. With this PR, the check may run on any team worker.

Reply:
Hmm are you sure about that? I think if there are 3 resource check containers on 3 different team workers, it won't end up using a team worker from a different team. For example, team-1 will never use the check container for the same scope on team-2 worker or team-3 worker. This is because we have a check in the `isWorkerCompatibleAndRunning` method that will check to see if the worker's `TeamID` is the same as the `TeamID` in the spec.

https://github.com/concourse/concourse/blob/72cb3661374529b6fb4ea80d98654f804e13a78d/atc/worker/pool.go#L421-L425

Reply:
@clarafu What you described is true with the existing logic, and that's the waste I mentioned. As the 3 resources share the same version history, it should not be a matter on which team's worker the check container runs, right? With this PR, the check container may run on any team's worker, there is only a single check container for the 3 resources, which saves 2 containers comparing to the existing logic.

Reply:
I do think it still matters which team worker the check container runs on because we make a guarantee that if you have a team worker, all workloads will be scheduled on it. Users might think its a bug when they see their checks selecting other team's workers.

Reply:
Same response as the last comment.